packages/core/src/services/builtInTools/runCode/withDependencies.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  BadRequestError,
+  LatitudeError,
+  PromisedResult,
+  Result,
+} from '../../../lib'
+import { CodeRunResult, CodeToolArgs, SupportedLanguage } from './types'
+import { normalizedResult, withSafeSandbox } from './sandbox'
+
+function getDependencyBuilder({ language }: { language: SupportedLanguage }) {
+  if (language === 'python') return (dep: string) => `pip install ${dep}`
+  if (language === 'javascript') return (dep: string) => `npm install ${dep}`
+
+  throw new BadRequestError(`Unsupported language: ${language}`)
+}
+
+function getFilename({ language }: { language: SupportedLanguage }) {
+  if (language === 'python') return 'index.py'
+  if (language === 'javascript') return 'index.js'
+  throw new BadRequestError(`Unsupported language: ${language}`)
+}
+
+function getRunCommand({
+  language,
+  filename,
+}: {
+  language: SupportedLanguage
+  filename: string
+}) {
+  if (language === 'python') return `python ${filename}`
+  if (language === 'javascript') return `node ${filename}`
+  throw new Error(`Unsupported language: ${language}`)
```

Comment:
What type of run error message generates this? Do we create a log with error result in this case?

Reply:
Got it